hack/update-generated-kms.sh

```diff
@@ -25,5 +25,6 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 # If the output set is changed update that function.
 
 "${KUBE_ROOT}/build/run.sh" hack/update-generated-kms-dockerized.sh "$@"
+APIROOTS="k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/v2alpha1" hack/update-generated-protobuf.sh "$@"
```

Comment:
this looks weird... was this commit intentional?

Reply:
This commit was intentional to generate https://github.com/kubernetes/kubernetes/pull/111126/files#diff-2d21114fa1edb2072dbaca8f208898f2b87453230424170e7996b4873d4146d1. 

`hack/update-generated-protobuf.sh` skips these nested dirs without using APIROOTS to override? I added it to the script here, so have all the generate commands in one place.

Reply:
this looks like it's running the REST API proto generators on the KMS go type? is that really what we want? if we don't need to identify arbitrary type/version data, would it make more sense to define the API in a .proto file and run the standard proto generator on it (adding another line in `hack/update-generated-kms-dockerized.sh` to do so)?

Reply:
We need to generate the `Marshal` and `Unmarshal` methods for the `EncryptedObject` that's used for serialization in the transformer interface. If we just define a `.proto` file and generate using protoc, we would only get `XXX_Unmarshal` and `XXX_Marshal` which we don't want to use directly?

Reply:
if we generate EncryptedObject like a normal proto message, and use proto.Marshal/Unmarshal, wouldn't that work?

```diff
diff --git a/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go b/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go
index c2b7d442441..c88fa5d95a1 100644
--- a/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go
+++ b/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go
@@ -26,6 +26,7 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/gogo/protobuf/proto"
 	"k8s.io/apimachinery/pkg/util/uuid"
 	"k8s.io/apiserver/pkg/storage/value"
 	kmstypes "k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/v2alpha1"
@@ -209,13 +210,13 @@ func (t *envelopeTransformer) getTransformer(encKey []byte) value.Transformer {
 
 // doEncode encodes the EncryptedObject to a byte array.
 func (t *envelopeTransformer) doEncode(request *kmstypes.EncryptedObject) ([]byte, error) {
-	return request.Marshal()
+	return proto.Marshal(request)
 }
 
 // doDecode decodes the byte array to an EncryptedObject.
 func (t *envelopeTransformer) doDecode(originalData []byte) (*kmstypes.EncryptedObject, error) {
 	o := &kmstypes.EncryptedObject{}
-	if err := o.Unmarshal(originalData); err != nil {
+	if err := proto.Unmarshal(originalData, o); err != nil {
 		return nil, err
 	}
 
```